### PR TITLE
parallel_global_det_atmos

### PR DESCRIPTION
**Pull request by PerryShafran-NOAA**
## Pull Request Testing ##

- [ ] Describe testing already performed for this Pull Request:</br>

These changes have been running in parallel in the emc.vpppg group account for awhile and works well.  

- [ ] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

Run the scripts as you normally would.  Check the output to compare to your own runs.  You may need to link to the /lfs/h2/emc/vpppg/noscrub/emc.vpppg/evs/v1.0 output if your runs need several days worth of data.  You may recommend merging into the auth repo but you do not have the authority to perform the merge.  

- [ ] Has the code been checked to ensure that no errors occur during the execution? **[Yes or No]**

- [ ] Do these updates/additions include sufficient testing updates? **[Yes or No]**

- [ ] Please complete this pull request review by **[Fill in date]**.</br>

At least 5 business days from receipt of this email.

## Pull Request Checklist ##

- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR description above.
- [ ] Ensure the PR title matches the feature branch name.
- [ ] Check the following:
- [ ]  Instructions provided on how to run
- [ ]  Developer's name is replaced by ${user} where necessary throughout the code
- [ ]  Check that the ecf file has all the proper definitions of variables
- [ ]  Check that the jobs file has all the proper settings of COMIN and COMOUT and other input variables
- [ ]  Check to see that the output directory structure is followed
- [ ]  Be sure that you are not using MET utilities outside the METplus wrapper structure

- [ ] After submitting the PR, select **Development** issue with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue.

### CODE REVIEW

**Comment by AliciaBentley-NOAA** [comment]
Apologies if this is included/not needed and I'm missing it, but are there
changes to run.ver and machine.conf that should be included in this PR?

___________________________________
*Alicia M. Bentley, Ph.D.*
Physical Scientist | Model Evaluation Group
<https://www.emc.ncep.noaa.gov/users/meg/home/>
Global Model Verification Team Lead
NOAA/NWS/NCEP/EMC (VPPPG Branch)
Work Phone: ‪(301) 683-3783‬

**Comment by malloryprow** [comment]
Testing should be done under "VERF-DEV"? We want to keep "EVS-DEV" for the parallel to track resources correct?

**Comment by PerryShafran-NOAA** [comment]
There shouldn't be any changes to either run.ver or machine.conf.

**Comment by PerryShafran-NOAA** [comment]
@malloryprow - yes, good catch!  I missed that.  Do your testing in VERF-DEV.  I'll be sure future PRs already have that changed back to VERF-DEV.

**Comment by malloryprow** [comment]
@PerryShafran-NOAA Okay! I'm getting things set up now. I don't know if I can get everything tested before tomorrow's production switch, but Friday or Monday at the latest I should have the testing done.

**Comment by PerryShafran-NOAA** [comment]
That's fine!  I'm off on Thursday aft/Friday, so whenever is good.

**Comment by LoganDawson-NOAA** [comment]
@PerryShafran-NOAA earlier this morning, you said:
> There shouldn't be any changes to either run.ver or machine.conf.

I see machine.conf listed in the file changes for PRs #66 #67 and #68 . And it looks like the MET_BIN_DIR setting in the branches associated with those PRs is different from the MET_BIN_DIR setting in the branch associated with this PR.

This looks like an issue that needs to be resolved before merging any of these PRs.

**Comment by PerryShafran-NOAA** [comment]
@LoganDawson-NOAA Thanks for bringing this up.  

That said, is it better for machine.conf to explicitly define MET_BIN_DIR as the following?

`MET_BIN_DIR = {MET_INSTALL_DIR}/bin`

Or shall we have it defined in there as:

`MET_BIN_DIR = {MET_INSTALL_DIR}/{ENV[MET_bin_exec]}`

and have everyone define MET_bin_exec somewhere in their scripts?  I'll ensure that whichever it is, it'll be consistent across EVS.

Perry

**Comment by LoganDawson-NOAA** [comment]
@PerryShafran-NOAA I think using the extra environment variable is preferable for now.

`MET_BIN_DIR = {MET_INSTALL_DIR}/{ENV[MET_bin_exec]}`

This allows us to more easily change the setting based on where the executables are located in installations on different supercomputers. (I think the METplus team has separately attempted to account for the variation between /bin and /exec, but I'm not sure if we're ready to adopt their approach yet)

**Comment by PerryShafran-NOAA** [comment]
Sounds good.  I'll make all those changes where necessary.  

Perry

File: ecf/global_det/stats/jevs_gfs_atmos_grid2grid_stats.ecf
Comment (malloryprow) [inline ecf/global_det/stats/jevs_gfs_atmos_grid2grid_stats.ecf:1]: Why was the resource changed from 2 to 1?

File: ecf/global_det/stats/jevs_gfs_atmos_grid2obs_stats.ecf
Comment (malloryprow) [inline ecf/global_det/stats/jevs_gfs_atmos_grid2obs_stats.ecf:1]: Why was the resource changed from 2 to 1?

**Comment by malloryprow** [comment]
**✔ global_det atmos prep and stats jobs that ran for 20230315 are good!**

I put them in a cron so they will continue to run over the weekend. On Monday I can review and run the global_det atmos plots jobs.

**Comment by malloryprow** [comment]
**✔ global_det atmos plots jobs that ran for 20230319 are good!**

Once the issues of ecf/global_det/stats/jevs_gfs_atmos_grid2grid_stats.ecf and ecf/global_det/stats/jevs_gfs_atmos_grid2obs_stats.ecf are fixed, this will be good to go.

**Comment by PerryShafran-NOAA** [comment]
Can you review these stats issues that you are speaking of?  I'm not sure I recall any being mentioned?

Thanks!

Perry

**Comment by malloryprow** [comment]
If you scroll up  a little bit on the pull request you will see the two reviews I opened about the files.

**Comment by PerryShafran-NOAA** [comment]
Is that in this PR?  I can't find those comments.

**Comment by malloryprow** [comment]
Screenshot from 2023-03-20 11-01-22

**Comment by PerryShafran-NOAA** [comment]
Oh!  I'm not entirely sure how that happened.  I'll check to see if those changes are running that way, and if not, make corrections and then re-upload the changes to github.

Perry

**Comment by malloryprow** [comment]
Sounds good. It doesn't seem to be impacting anything and the final .stat files are produced. I had bumped up the resources at one point to decrease run time.

**Review by malloryprow — APPROVED** [review]

File: ecf/global_det/stats/jevs_gfs_atmos_grid2grid_stats.ecf
Comment (PerryShafran-NOAA) [reply ecf/global_det/stats/jevs_gfs_atmos_grid2grid_stats.ecf:1]: Resource changed back to 2.  Do you want to re-run again to test?

File: ecf/global_det/stats/jevs_gfs_atmos_grid2obs_stats.ecf
Comment (PerryShafran-NOAA) [reply ecf/global_det/stats/jevs_gfs_atmos_grid2obs_stats.ecf:1]: Resource changed back to 2.  Do you want to re-run again to test?

**Review by malloryprow — APPROVED** [review]
All looks good!

File: ecf/global_det/stats/jevs_gfs_atmos_grid2obs_stats.ecf
Comment (malloryprow) [reply ecf/global_det/stats/jevs_gfs_atmos_grid2obs_stats.ecf:1]: I think it is good. It was still running fine before.

File: ecf/global_det/stats/jevs_gfs_atmos_grid2grid_stats.ecf
Comment (malloryprow) [reply ecf/global_det/stats/jevs_gfs_atmos_grid2grid_stats.ecf:1]: I think it is good. It was still running fine before.

**Review by malloryprow — APPROVED** [review]

**Comment by PerryShafran-NOAA** [comment]
@malloryprow - just to confirm, we are good with merging with the changes to put the resources back to where they should be in those two scripts?

**Comment by malloryprow** [comment]
Yup!

**Comment by PerryShafran-NOAA** [comment]
@ShelleyMelchior-NOAA It seems that as the person who submitted this PR, I cannot also be an approver.  It needs to be approved by someone with write access, and that would be you.  Mallory has approved all changes of this parallel to be merged to the auth repo, it just needs your approval and then you can merge.  I think you'll need to do this for all of the parallel PRs that I have created.

**Review by ShelleyMelchior-NOAA — APPROVED** [review]
I gave a quick review of all files changed and all looks good. I am approving this PR.

**Pull request merged**
